java-client/session/src/main/java/io/deephaven/client/impl/HasTypedTicket.java

```diff
@@ -0,0 +1,14 @@
+/**
+ * Copyright (c) 2016-2023 Deephaven Data Labs and Patent Pending
+ */
+package io.deephaven.client.impl;
+
+public interface HasTypedTicket extends HasTicketId {
```

Comment:
Given that this is explicitly for the low level "i already know how the underlying impl and wire format works" api, whats the purpose in interfaces like this, that only have a single implementation? I get it for the api's internals with multiple impls, or exposing types through the high level api with a low level impl (with hidden other details), but I don't see the benefit here.

Reply:
I don't necessarily see `HasTypedTicket` as exclusively "low-level".

I think it's reasonable from a high-level api to do something like:

```java
objectService.fetchable(new ScopeId("my_custom_object").ticketId().toTypedTicket("my_object_type"))
```

(excuse the un-ergonimics of it; we _can_ make it better...)

In this case, the user knows they have a globally scoped variable named "my_custom_object" of custom type "my_object_type", and they didn't need to know the low-level wire details beyond that. 

Generally, I'm happy to make strides in the java client code to better hide abstractions the user's don't / shouldn't need to know about.